primefaces/src/main/java/org/primefaces/util/ResourceUtils.java

```diff
@@ -335,4 +320,197 @@ public static Resource evaluateResourceExpression(FacesContext facesContext, Val
+        UIOutput outputScript = new UIOutput();
+        outputScript.setRendererType(type);
+        return outputScript;
+    }
+
+    /**
+     * Adds a resource to the head section.
+     */
+    public static void addResourceToHead(FacesContext context, String type, String libraryName, String resourceName) {
+        addScriptResourceToTarget(context, type, libraryName, resourceName, "head");
+    }
+
+    /**
+     * Adds a resource to the body section.
+     */
+    public static void addResourceToBody(FacesContext context, String type, String libraryName, String resourceName) {
+        addScriptResourceToTarget(context, type, libraryName, resourceName, "body");
+    }
+
+    /**
+     * Adds a component resource to the specified target after ensuring it has a unique ID.
+     *
+     * @param context The FacesContext
+     * @param resource The component resource to add
+     * @param target The target location ("head" or "body")
+     * @return The added component
+     */
+    private static UIComponent addComponentResource(FacesContext context, UIComponent resource, String target) {
+        if (resource.getId() == null) {
+            setComponentResourceUniqueId(context, resource);
+        }
+
+        context.getViewRoot().addComponentResource(context, resource, target);
+        return resource;
+    }
+
+    /**
+     * Adds inline JavaScript code to the body section.
+     *
+     * @param context The FacesContext
+     * @param script The JavaScript code to add
+     */
+    public static void addScriptToBody(FacesContext context, String script) {
+        addScriptToTarget(context, script, "body");
+    }
+
+    /**
+     * Adds inline JavaScript code to the head section.
+     *
+     * @param context The FacesContext
+     * @param script The JavaScript code to add
+     */
+    public static void addScriptToHead(FacesContext context, String script) {
+        addScriptToTarget(context, script, "head");
+    }
+
+    /**
+     * Adds inline JavaScript code to the specified target section.
+     *
+     * @param context The FacesContext
+     * @param script The JavaScript code to add
+     * @param target The target location ("head" or "body")
+     */
+    public static void addScriptToTarget(FacesContext context, String script, String target) {
+        UIOutput outputScript = createResource(RENDERER_SCRIPT);
+        UIOutput content = new UIOutput();
+        content.setValue(script);
+        outputScript.getChildren().add(content);
+        addComponentResource(context, outputScript, target);
+    }
+
+    /**
+     * Sets a unique ID for a component resource, handling MyFaces-specific ID generation. MyFaces requires a special flag to be set during ID generation for
+     * resources.
+     *
+     * @param context The FacesContext
+     * @param resource The component resource needing a unique ID
+     */
+    public static void setComponentResourceUniqueId(FacesContext context, UIComponent resource) {
```

Comment:
why is this necessary at all?
in the passt we didnt set a ID in this methods and it worked fine?

Reply:
in general such impl specific hacks are very dangerous and we would need to check it in myfaces first

Reply:
Agreed I grabbed this directly from OmniFaces as this is how BalusC was doing it. Also a note our PF Extensions code had this as well..

```java
        // https://github.com/primefaces-extensions/primefaces-extensions/issues/517
        // MyFaces needs ID set to prevent duplicate ID check in Dev mode, Mojarra does not
        final PrimeEnvironment environment = PrimeApplicationContext.getCurrentInstance(context).getEnvironment();
        if (!environment.isMojarra()) {
            js.setId("primfaces-script-processor");
        }
```

I worked with @volosied in MyFaces trying to resolve this a a couple of years ago.  We had a long back and forth here about it: https://github.com/primefaces-extensions/primefaces-extensions/issues/517